dali/daliadmin/daliadmin.cpp

```diff
@@ -3219,7 +3239,135 @@ void testThorRunningWUs()
+    Owned<IPropertyTreeIterator> varIter = variables->getElements("*");
+    StringBuffer nameBase;
+    ForEach(*varIter)
+    {
+        IPropertyTree &variable = varIter->query();
+        if (streq("Variable", variable.queryName()))
+        {
+            const char *name = variable.queryProp("@name");
+            const char *lfn = name;
+            if (RE.find(name))
+            {
+                RE.substitute(nameBase.clear(), "#1");
+                lfn = nameBase.str();
+            }
+            auto it = varMap.find(lfn);
+            if (it == varMap.end())
+                it = varMap.insert({ lfn, {} }).first;
+            auto &e = it->second;
+            e.push_back(&variable);
+        }
+        else if (reconstruct)
+            newVariables->addPropTree(variable.queryName(), LINK(&variable));
+    }
+    unsigned total = (unsigned)varMap.size();
+    PROGLOG("Found %u global workunit entries. Time taken: %u ms", total, timer.elapsedMs());
+
+    timer.reset();
+    unsigned checked = 0, deletes = 0, auxDeletes = 0, existingPersists = 0;
+    cycle_t connCycles = 0, removeCycles = 0;
+    CCycleTimer per100Timer;
+
+    bool removePTree = true;
+    std::vector<const IPropertyTree *> varsToKeep;
+    auto varMapIt = varMap.begin();
+    StringBuffer lfnXpath;
+    while (varMapIt != varMap.end())
+    {
+        const char *name = varMapIt->first.c_str();
+
+        try
+        {
+            CDfsLogicalFileName lfn;
+            lfn.set(name);
+            lfn.makeFullnameQuery(lfnXpath.clear(), DXB_File);
+
+            Owned<IRemoteConnection> fConn = querySDS().connect(lfnXpath, myProcessSession(), RTM_LOCK_READ, daliConnectTimeoutMs);
+            if (!fConn) // doesn't exist, clear up Variable and associates
+            {
+                ++deletes;
+
+                if (reconstruct)
+                    varMapIt = varMap.erase(varMapIt);
+                else
+                {
+                    for (auto &t: varMapIt->second)
+                    {
+                        if (!dryrun)
+                            verifyex(variables->removeTree(t));
+                        ++auxDeletes;
+                    }
+                    varMapIt++;
+                }
+            }
+            else
+            {
+                ++existingPersists;
+                varMapIt++;
+            }
+        }
+        catch (IException *e)
+        {
+            VStringBuffer errMsg("Skpping: %s", name);
+            EXCLOG(e, errMsg.str());
+            e->Release();
+        }
+        ++checked;
+
+        if (0 == (checked % reportInterval))
+        {
+            cycle_t perVarCycles = per100Timer.elapsedCycles() / reportInterval;
+
+            per100Timer.reset();
+            PROGLOG("%sChecked: [%u / %u] - deletes: %u, auxDeletes: %u, persists: %u. [Avg ms: %2.2f]", dryrun?"DRYRUN:":"", checked, total, deletes, auxDeletes, existingPersists, static_cast<unsigned>(cycle_to_microsec(perVarCycles))/1000.0);
```

Comment:
minor: I think the reported average is in seconds, rather than ms (since you're dividing ms by 1000.0)

Reply:
ignore that.  Confused microseconds and milli seconds

Reply:
it's microsecs / 1000.0.
So millionths of sec.  / 1000.0 => millisecs ?